src/cfe.c

```diff
@@ -559,32 +606,38 @@ void Xinanjiang_partitioning_scheme(double water_input_depth_m, double field_cap
   if(max_free_water_m < free_water_m) free_water_m = max_free_water_m;
   if(max_tension_water_m < tension_water_m) tension_water_m = max_tension_water_m;
 
-  // NOTE: the impervious surface runoff assumptions due to frozen soil used in NWM 3.0 have not been included.
-  // We are assuming an impervious area due to frozen soils equal to 0 (see eq. 309 from Knoben et al).
-
-  // The total (pervious) runoff is first estimated before partitioning into surface and subsurface components.
-  // See Knoben et al eq 310 for total runoff and eqs 313-315 for partitioning between surface and subsurface
-  // components.
+   // estimate the fraction of the modeled area that is impervious (impervious_fraction) based on 
+  // urban classification (hard coded 95% [0.95] impervious) and frozen soils (passed to cfe 
+  // from freeze-thaw model) using a weighted average. 
+  impervious_fraction = (parms->urban_decimal_fraction * 0.95) + 
+			((1 - parms->urban_decimal_fraction) * ice_fraction_xinan);
+  
+  // Calculate the impervious runoff (see eq. 309 from Knoben et al).
+  impervious_runoff_m = impervious_fraction * water_input_depth_m;
 
   // Calculate total estimated pervious runoff. 

```

Comment:
Remove the duplicate comment and note in lines 618-620:
  // Calculate total estimated pervious runoff. 
  // NOTE: If the impervious surface runoff due to frozen soils is added,
  // the pervious_runoff_m equation will need to be adjusted by the fraction of pervious area.

Reply:
removed